virttest/cpu.py

```diff
@@ -1258,3 +1261,110 @@ def get_qemu_best_cpu_model(params):
             return host_cpu_model
     # If no host cpu model can be found on qemu_cpu_models, choose the default
     return params.get("default_cpu_model", None)
+
+
+def get_qemu_best_cpu_info(params):
+    """
+    Get the best cpu model and features available for qemu
+    """
+    cpu_info = {}
+    cpu_model = get_qemu_best_cpu_model(params)
+    # Expand if needed
+    cpu_flags = ""
+    cpu_info["model"] = cpu_model
+    cpu_info["flags"] = cpu_flags
+    return cpu_info
+
+
+def convert_flags_from_str_to_dict(cpu_flags_str):
+    """
+    Convert cpu flags string to a dict
+    """
+    flags_dict = {}
+    if cpu_flags_str:
+        flags_list = cpu_flags_str.lstrip(",").split(",")
+        for item in flags_list:
+            if item.startswith("+"):
+                flags_dict[item[1:]] = "on"
+            elif item.startswith("-"):
+                flags_dict[item[1:]] = "off"
+            elif item.find("=") != -1:
+                item = item.split("=")
+                flags_dict[item[0]] = item[1]
+            else:
+                flags_dict[item] = None
+    return flags_dict
+
+
+def get_cpu_info_from_qemu_cli(name):
```

Comment:
It is only one step of `get_cpu_info_from_virsh`, they are not same level,  suggest move into `get_cpu_info_from_virsh` or change the function name.

Reply:
It can be used by tp-qemu for the new case. What do you mean by changing the function name?

Reply:
> It can be used by tp-qemu for the new case. What do you mean by changing the function name?

Does this means tp-qemu skip `get_cpu_info_from_virsh`, use specified xml and define/start/destroy/undefine vm in test case level ?

Reply:
I'm not sure. If use `get_cpu_info_from_virsh` directly, it will need to run virsh define/start twice.  @nanliu-r Could you please share your thoughts?

Reply:
> I'm not sure. If use `get_cpu_info_from_virsh` directly, it will need to run virsh define/start twice. @nanliu-r Could you please share your thoughts?

Yes, tp-qemu can use this function directly while developing the cpu_model case, we can keep it and change the name, thanks.

Reply:
> > I'm not sure. If use `get_cpu_info_from_virsh` directly, it will need to run virsh define/start twice. @nanliu-r Could you please share your thoughts?
> 
> Yes, tp-qemu can use this function directly while developing the cpu_model case, we can keep it and change the name, thanks.

Except ` <cpu mode='host-model'/>`,  to test other cpu model with virsh in tp-qemu.
So from my point of view, it is a new requirement that didn't be raised before,   make cpu model configurable while host-model is the default in avocado-vt.

cc @luckyh

Reply:
I don't get your point. I just wrapped some steps into a function, which happens to be able to be used a tp-qemu case. What's the new requirement?

Reply:
My understanding for @nanliu-r 's comment is,  there are requirement to use virsh to test other cpu module and get cpu info for qemu cli. 

IMO, to resue `get_cpu_info_from_qemu_cli` in tp-qemu,  `get_cpu_info_from_virsh` need be rewrite in tp-qemu as different requirement. Correct me if my understanding is wrong.

Reply:
I think there is a misunderstanding here. The case needs to dumpxml and parse xml, apart from that, other steps like virsh define/start, get_cpu_info_from_qemu_cli etc, are same.
Let's put aside the tp-qemu case. If we prefer to keep this function, what kind of function name do you want to change to?

Reply:
> My understanding for @nanliu-r 's comment is, there are requirement to use virsh to test other cpu module and get cpu info for qemu cli.
> 
> IMO, to resue `get_cpu_info_from_qemu_cli` in tp-qemu, `get_cpu_info_from_virsh` need be rewrite in tp-qemu as different requirement. Correct me if my understanding is wrong.

I want to say that it just is convenient to reuse it but this doesn't mean that the function is designed for tp-qemu reuse.

Reply:
Ok for me if no other cpu model be involved.  Test case response for generate xml, define/start/... , use the vt funtion to get cpu info.
How about  `get_cpu_info_from_virsh_qemu_cli` ?  One option for virsh,  (same level with something like `get_cpu_info_from_virsh_xml`)

Reply:
Updated, pls help review again.